### DISCUSSION TITLE
Failing build when deployed to vercel or netlify

### DISCUSSION THREAD

**Discussion by ExpressGradient** [vercel/next.js#26314] [Help] [answered]
### What version of Next.js are you using?

11.0.0

### What version of Node.js are you using?

14.15.0

### What browser are you using?

Firefox Developer Edition

### What operating system are you using?

Windows 10

### How are you deploying your application?

Vercel and Netlify

### Describe the Bug

I have a blog built with Next.js and MongoDB.
It's Static Generated with ISR.
When I run build locally, everything works fine.
But when I deploy to vercel or netlify, build fails while generating the latest page.
The error is 
```
Error occurred prerendering page "/blog/<blog_name>". Read more: https://nextjs.org/docs/messages/prerender-error
```
also
```
> Build error occurred
Error: Export encountered errors on following paths:
/blog/<blog_name>
```

### Expected Behavior

Expected behavior is to deploy without no problems. Since everything worked fine locally.

### To Reproduce

You can clone my [github repo](https://github.com/ExpressGradient/praneeth) and deploy it to reproduce the failing build error.

**Comment by timneutkens** [comment]
https://github.com/ExpressGradient/praneeth/blob/main/pages/blog/%5Bslug%5D.tsx#L56
https://github.com/ExpressGradient/praneeth/blob/main/pages/blog/%5Bslug%5D.tsx#L60

You likely return a path named `<blog_name>`

**Reply by ExpressGradient** [reply]
Sorry, if I explained myself wrong, the `<blog_name>` is just a placeholder for the blog slug.
Also there is no problem with the code, since it is working perfectly fine on mine and my friend's machine locally. Even you can clone and try.
There are errors only when I deploy it.
That too error generating only one page out of 8 pages.
Rest other pages are built fine.

**Reply by adebiyial** [reply]
I understand what's going on.

Fetching data with one of the `get... ` functions in a dynamic page/route requires that you create a fallback in case the data is null or undefined.

In this case you try to access properties on the `blog` object but you need to be aware that it can be null or undefined. So you should conditionally render the places where `blog` is used.

```
... 
if(blog) return {...}
return {...}
```

**Reply by ExpressGradient** [reply]
If that's the case, my only question I'm struggling with is, why would it work locally, it should show up errors on my machine too. But nope, all pages are built fine.
It even worked on my friend's PC.
Can you please clone the repo, provide a mongo cluster key and try building it?
It would help me a lot to make my case. It's been 2 days since the site's been down.
Thanks.

**Reply by adebiyial** [reply]
I'd like to know why it works locally too but not when deployed. I guess it's a Vercel thing, not Next.js. I also learned about in the same scenario.

I'd run your code but follow my guide and redeploy. Let's see if that works. Also, I'd try to deploy same thing to Netlify and see the reaction.

I believe `blog` will be `undefined` if you closely check the CI/CD log.

**Reply by ExpressGradient** [reply]
Thank You.

**Comment by ExpressGradient** [comment, marked as answer]
I have solved the issue by adding a cached Db
```ts
let cachedDb: Db = null;

export const connectDb = async (dbName: string): Db => {
    if(cachedDb) {
        return cachedDb;
    }

    await client.connect();
    cachedDb = client.db(dbName);
    return cachedDb;
}
```

**Reply by adebiyial** [reply]
Ah. Great!
